core/iwasm/interpreter/wasm_loader.c

```diff
@@ -1692,8 +1692,16 @@ init_function_local_offsets(WASMFunction *func,
     uint64 total_size = sizeof(uint16) * ((uint64)param_count + local_count);
 
     if (!(func->local_offsets =
-                loader_malloc(total_size, error_buf, error_buf_size))) {
-        return false;
+            loader_malloc(total_size, error_buf, error_buf_size))) {
+        /*
+         * When total_size is 0, the return value of malloc(0)
+         * is NULL on some platform.
+         * Note: If size is 0, then malloc() returns either NULL,
+         * or a unique pointer value that can later be successfully passed to free().
+         */
+        if (total_size != 0) {
+            return false;
+        }
```

Comment:
How about changing to:
```C
        if (total_size > 0
            && !(func->local_offsets =
                 loader_malloc(total_size, error_buf, error_buf_size))) {
            return false;
        }
```
If total size is 0, just leave it NULL.

Reply:
Done